### DISCUSSION TITLE
How to convert segmentation labels to a map of pixel coordinates to identified class?

### DISCUSSION THREAD

**Discussion by SemperZero** [roboflow/notebooks#76] [Q&A] [answered]
example label: `2 0.0125 0.517187 0.0104167 0.51875 0.0104167 0.523438 0.00833333 0.525 0.00833333 0.7875 0.0104167 0.789062 0.0104167 0.803125 0.0125 0.804688 0.0145833 0.803125 0.0208333 0.803125 0.0229167 0.801562 0.025 0.801562 0.0270833 0.8 0.0291667 0.8 0.0333333 0.796875 0.0333333 0.795313 0.03125 0.79375 0.03125 0.790625 0.0291667 0.789062 0.0291667 0.785937 0.0270833 0.784375 0.0270833 0.78125 0.025 0.779688 0.025 0.765625 0.0270833 0.764063 0.0270833 0.754687 0.0291667 0.753125 0.0291667 0.739062 0.03125 0.7375 0.03125 0.71875 0.0333333 0.717188 0.0333333 0.704687 0.0354167 0.703125 0.0354167 0.701563 0.0375 0.7 0.0395833 0.7 0.0416667 0.698438 0.05 0.698438 0.0541667 0.701563 0.0541667 0.709375 0.05625 0.710938 0.05625 0.71875 0.0583333 0.720312 0.0583333 0.728125 0.0604167 0.729688 0.0604167 0.734375 0.0625 0.735937 0.0625 0.7375 0.0645833 0.739062 0.0645833 0.740625 0.06875 0.74375 0.06875 0.746875 0.0729167 0.75 0.08125 0.75 0.0895833 0.74375 0.0895833 0.742188 0.09375 0.739062 0.09375 0.7375 0.0958333 0.735937 `

how can i convert this to a map like:

{(1, 100) : 2,
(2, 100): 2,
(3, 101): 2,
...
(100, 150): 2}

i need it so that i can take the information i have from a synchronized depth camera and compute the average depth of an object. i'd take each pixel coordinate from the polygon encoded by the label and then take its depth from the camera image at the same pixel, but i need to know how to decode it.

also, how do i take the label info in code? `results[i].masks` and `results[i].boxes` seem to print different information compared to the label written in the log files

**Comment by SemperZero** [comment, marked as answer]
figured it on my own and submitted a PR. will clean it up in the following days:

https://github.com/ultralytics/ultralytics/pull/1233

**Comment by SkalskiP** [comment]
Great to hear that you could figure it out on your own. Sorry, it took me so long to answer. I have been working on a new YT video over the past few days.
